src/aiida/engine/processes/process.py

```diff
@@ -415,8 +415,19 @@ def on_create(self) -> None:
     @override
     def on_entered(self, from_state: Optional[plumpy.process_states.State]) -> None:
         """After entering a new state, save a checkpoint and update the latest process state change timestamp."""
+        from plumpy import ProcessState
+
         from aiida.engine.utils import set_process_state_change_timestamp
 
+        super().on_entered(from_state)
```

Comment:
What does the parent class version of this method do? Why was it called at the end of the method first and is there any risk of moving it up?

Reply:
The parent class is the `Process` class in `plumpy` and above that the `StateMachine`. It makes sure that all state transitions are handled properly and callbacks that are registered are invoked. To be completely honest, I am not sure why it was below, I just put it back where it should be really. It may have been moved down together with the change relating to the huge comment of the `update_outputs` having to be called before updating the node state. But those are operations that work on the node, not the process itself, so don't see how that should affect the logic called in the parent `Process` class.